src/test/java/com/conveyal/datatools/manager/controllers/api/DeploymentControllerTest.java

```diff
@@ -0,0 +1,88 @@
+
+import java.io.IOException;
+
+import static com.conveyal.datatools.TestUtils.createFeedVersion;
+import static com.conveyal.datatools.TestUtils.zipFolderFiles;
+import static org.eclipse.jetty.http.HttpStatus.OK_200;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+
+
+public class DeploymentControllerTest extends DatatoolsTest {
+    private static Project project = null;
+    private static FeedSource feedSource = null;
+    private static FeedVersion feedVersion = null;
+    private static ObjectMapper objectMapper = new ObjectMapper();
+
+    @BeforeAll
+    public static void setUp() throws IOException {
+        DatatoolsTest.setUp();
+        Auth0Connection.setAuthDisabled(true);
+        project = new Project();
+        project.name = "ProjectOne";
+        Persistence.projects.create(project);
+        feedSource = new FeedSource("FeedSourceOne");
+        feedSource.projectId = project.id;
+        Persistence.feedSources.create(feedSource);
+        feedVersion = createFeedVersion(
+            feedSource,
+            zipFolderFiles("fake-agency-with-calendar-and-calendar-dates")
+        );
+    }
+
+    @AfterAll
+    public static void tearDown() {
+        Auth0Connection.setAuthDisabled(Auth0Connection.getDefaultAuthDisabled());
+        if (project != null) {
+            project.delete();
+        }
+    }
+
+    @Test
+    public void canCreateFeedSourceSpecificDeploymentWithDefaultRouter() throws IOException {
+        // Create a feed source specific deployment.
+        HttpResponse createDeploymentResponse = TestUtils.makeRequest( "/api/manager/secure/deployments/fromfeedsource/" + feedSource.id, 
+            null,
+            HttpUtils.REQUEST_METHOD.POST
+        );
+        Deployment deployment = objectMapper.readValue(createDeploymentResponse.getEntity().getContent(), Deployment.class);
+        
+        assertEquals(OK_200, createDeploymentResponse.getStatusLine().getStatusCode());
+        assertEquals(null, deployment.routerId ); 
+    }
+
+    @Test
+    public void canCreateFeedSourceSpecificDeploymentForNysdot() throws IOException {
+        String nysdotEnabledField = "extensions.nysdot.enabled";
+        String prevNysdotEnabled = DataManager.getConfigPropertyAsText(nysdotEnabledField);
+        DataManager.overrideConfigProperty(nysdotEnabledField, "true" );
+        HttpResponse createDeploymentResponse = TestUtils.makeRequest( "/api/manager/secure/deployments/fromfeedsource/" + feedSource.id, 
+            null,
+            HttpUtils.REQUEST_METHOD.POST
+        );
+        Deployment deployment = objectMapper.readValue(createDeploymentResponse.getEntity().getContent(), Deployment.class);
+
+        assertEquals(OK_200, createDeploymentResponse.getStatusLine().getStatusCode());
+        assertEquals( StringUtils.getCleanName(feedSource.name) + "_" + feedSource.id, deployment.routerId );
+
+        DataManager.overrideConfigProperty(nysdotEnabledField, prevNysdotEnabled);
```

Comment:
This change should be done after the test. If any of the above fails, this piece of code won't be reached.

Reply:
@evansiroky, could you elaborate for @philip-cline with a code snippet on how one might change this test to meet your requests about setup/teardown?

Reply:
I discussed in our company chat.

Reply:
Moved tests into separate classes to fix this. See latest commit https://github.com/ibi-group/datatools-server/pull/385/commits/e771e74bbc99f3db82288dda091d17cd6f765c51